sopel_modules/youtube/youtube.py

```diff
@@ -66,10 +85,16 @@ def configure(config):
 def setup(bot):
     bot.config.define_section('youtube', YoutubeSection)
     if 'youtube_api_client' not in bot.memory:
-        bot.memory['youtube_api_client'] = apiclient.discovery.build(
-            "youtube", "v3",
-            developerKey=bot.config.youtube.api_key,
-            cache_discovery=False)
+        reason = None
+        try:
+            bot.memory['youtube_api_client'] = googleapiclient.discovery.build(
+                "youtube", "v3",
+                developerKey=bot.config.youtube.api_key,
+                cache_discovery=False)
+        except googleapiclient.errors.HttpError as e:
```

Comment:
What happens if this fails for a different reason?

Reply:
An exception not caught bubbles up, so Sopel's normal error handler would log it as usual. Unless a non-HTTP error could also result in printing out the full URL with API key embedded, that should be fine. This is `setup()`, so it won't be possible to leak any error to IRC yet—Sopel's not connected at this stage.

Reply:
👍